### PR TITLE
[Bug Fix][Viewer] Fix inconsistency between mouse move and keyboard move

### PR DESCRIPTION
**Pull request by bigbike**
## Motivation and Context
Play will the viewer you will find:
-) Rendering images are discontinuous when you press left mouse button to rotate + scroll to zoom in and out, and use keyboard to control the agent and the sensors;
-) Rotation on left mouse button suffers significant numerical drifts, a bit difficult to manipulate the scene.

Root cause:
Different from keyboard event, when handling mouse press and scroll event, we convert such motions directly to the render camera, not to the agent or the attached sensors.

We are suffering such artifacts for quite a long time. This is the 1st diff to fix this problem.

## Types of changes

- [ ] Docs change / refactoring / dependency upgrade
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have completed my CLA (see **CONTRIBUTING**)
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: src/CMakeLists.txt [hunk 1]
```diff
@@ -65,6 +65,7 @@ if(BUILD_TEST)
 endif()
 
 # include source dirs
+message(STATUS "============ SOURCE ++++ " ${PROJECT_SOURCE_DIR})
```
Comment (bigbike) [inline src/CMakeLists.txt:68]: Weird: I did not add this line here.

File: src/utils/viewer/viewer.cpp [hunk 2]
```diff
@@ -586,20 +596,22 @@ void Viewer::mouseScrollEvent(MouseScrollEvent& event) {
 }
 
 void Viewer::mouseMoveEvent(MouseMoveEvent& event) {
+  if (mousePreviousPosition2D_ == Mn::Vector2i{-1})
+    mousePreviousPosition2D_ = event.position();
+
+  const Mn::Vector2i delta = event.position() - mousePreviousPosition2D_;
+  mousePreviousPosition2D_ = event.position();
```
Comment (mosra) [inline src/utils/viewer/viewer.cpp:603]: Both SDL and GLFW app provide a convenience [relativePosition()](https://doc.magnum.graphics/magnum/classMagnum_1_1Platform_1_1GlfwApplication_1_1MouseMoveEvent.html#a4954fc274c486ebac749fedc730cf710) that is meant to be used in this case.

The `mousePreviousPosition2D_` member is then not needed anymore.

File: src/utils/viewer/viewer.cpp [hunk 3]
```diff
@@ -134,8 +135,11 @@ class Viewer : public Mn::Platform::Application {
   esp::gfx::RenderCamera* renderCamera_ = nullptr;
   esp::nav::PathFinder::ptr pathfinder_;
   esp::scene::ObjectControls controls_;
+  // TODO: not needed anymore, will be depricated in next PR.
   Mn::Vector3 previousPosition_;
```
Comment (mosra) [inline src/utils/viewer/viewer.cpp:139]: remove now and not later? i don't see it used anymore

Comment (bigbike) [reply src/utils/viewer/viewer.cpp:139]: OK, as you wish.
I was planning to fix it in another diff.
This diff is to just fix exactly one problem and let everybody reach a consensus.

**Pull request merged**